src/http/KeyAcceptDialog.cpp

```diff
@@ -0,0 +1,112 @@
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation, either version 2 or (at your option)
+*  version 3 of the License.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "KeyAcceptDialog.h"
+#include "ui_KeyAcceptDialog.h"
+#include "core/Entry.h"
+#include <QStandardItemModel>
+#include <QStandardItem>
+#include <Qt>
+
+KeyAcceptDialog::KeyAcceptDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::KeyAcceptDialog())
+{
+    this->setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
+
+    ui->setupUi(this);
+    connect(ui->buttonBox, SIGNAL(accepted()), this, SLOT(accept()));
+    connect(ui->buttonBox, SIGNAL(rejected()), this, SLOT(reject()));
+
+    QStandardItemModel* listViewModel = new QStandardItemModel(ui->databasesListView);
+    ui->databasesListView->setModel(listViewModel);
+}
+
+KeyAcceptDialog::~KeyAcceptDialog()
+{
+}
+
+void KeyAcceptDialog::setItems(const QList<QString> &items)
+{
+    QStandardItemModel* listViewModel = qobject_cast<QStandardItemModel*>(ui->databasesListView->model());
+
+    listViewModel->clear();
+
+    for (QString item: items) {
+        QStandardItem* listItem = new QStandardItem(item);
+        listItem->setFlags(Qt::ItemIsEnabled | Qt::ItemIsUserCheckable);
+        listItem->setData(Qt::Unchecked, Qt::CheckStateRole);
+
+        listViewModel->appendRow(listItem);
+    }
+}
+
+void KeyAcceptDialog::setItemEnabled(int itemIndex, bool enabled)
+{
+    QStandardItemModel* listViewModel = qobject_cast<QStandardItemModel*>(ui->databasesListView->model());
+
+    QStandardItem* item = listViewModel->item(itemIndex);
+
+    if (item) {
+        item->setEnabled(enabled);
+    }
+}
+
+void KeyAcceptDialog::setItemChecked(int itemIndex, bool checked)
+{
+    QStandardItemModel* listViewModel = qobject_cast<QStandardItemModel*>(ui->databasesListView->model());
+
+    QStandardItem* item = listViewModel->item(itemIndex);
+
+    if (item) {
+        if (checked) {
+            item->setCheckState(Qt::Checked);
+        } else {
+            item->setCheckState(Qt::Unchecked);
+        }
+    }
+}
+
+QList<int> KeyAcceptDialog::getCheckedItems()
+{
+    QStandardItemModel* listViewModel = qobject_cast<QStandardItemModel*>(ui->databasesListView->model());
+
+    int numRows = listViewModel->rowCount();
+
+    //XXX Memory Leak
+    QList<int> resultList;
+
+    for (int row = 0; row < numRows; row++) {
+        QStandardItem* item = listViewModel->item(row);
+        if (item->checkState() == Qt::Checked) {
+            resultList << row;
+        }
+    }
+
+    return resultList;
```

Comment:
Is it okay to stack-allocate (? - I am not sure if that is what happens, and whether Qt transforms that into something else) `resultList` here and then return it to the caller, where the caller will simply let it go out of scope?

I.e. will the object be destroyed once it runs out of scope in the caller, and is this the proper way to do it or should this be done differently?

Reply:
Answered on second question